oxidation/bindings/electron/src/index.d.ts

```diff
@@ -37,4 +37,19 @@ export type DeviceFileType =
   | DeviceFileTypeSmartcard
 
 
+// LoggedCoreError
+export interface LoggedCoreErrorDisconnected {
+    tag: "Disconnected"
+}
+export interface LoggedCoreErrorInvalidIndex {
+    tag: "InvalidIndex"
+    index: number;
+}
+export type LoggedCoreError =
+  | LoggedCoreErrorDisconnected
+  | LoggedCoreErrorInvalidIndex
+
+
 export function listAvailableDevices(path: string): Promise<Array<AvailableDevice>>;
+export function login(test_device_id: string): Promise<number>;
```

Comment:
Can this function be typed like so ?

```suggestion
export async function login(test_device_id: string): number;
```

Reply:
Seems `js dev` use `Promise` in `vue`

Reply:
from what I understand, ts requires that an async function returns a Promise (cf. https://github.com/Scille/parsec-cloud/blob/90ad6abc6e6c4810d43a6ab75eb101ba6d7ea4fa/oxidation/client/src/plugins/libparsec/trampoline-web.ts.in#L1)

Not sure if there is a difference between `async function foo(): Promise<number>` and `function foo(): Promise<number>` though...

Reply:
:shrug: